templates/named.view_header.erb

```diff
@@ -0,0 +1,55 @@
+view "<%= @title %>" {
+<% unless @forwarders.empty? -%>
+    forward <%= @forward %>;
+    forwarders { <%= @forwarders.join('; ') %>; };
+<% end -%>
+<% unless @match_clients.empty? -%>
+    match-clients { <%= @match_clients.join('; ') %>; };
+<% end -%>
+<% unless @match_destinations.empty? -%>
+    match-destinations { <%= @match_destinations.join('; ') %>; };
+<% end -%>
+<% unless @allow_transfer.empty? -%>
+    allow-transfer { <%= @allow_transfer.join('; ') %>; };
+<% end -%>
+<% unless @allow_recursion.empty? -%>
+    allow-recursion { <%= @allow_recursion.join('; ') %>; };
+<% end -%>
+<% unless @allow_query.empty? -%>
+    allow-query { <%= @allow_query.join('; ') %>; };
+<% end -%>
+<% unless @allow_query_cache.empty? -%>
+    allow-query-cache { <%= @allow_query_cache.join('; ') %>; };
+<% end -%>
+<% unless @also_notify.empty? -%>
+    also-notify { <%= @also_notify.join('; ') %>; };
+<% end -%>
+<% if @dns_notify -%>
+    notify <%= @dns_notify %>;
+<% end -%>
+<% if @match_recursive_only -%>
+    match-recursive-only <%= @match_recursive_only %>;
+<% end -%>
+<% if @recursion -%>
+    recursion <%= @recursion %>;
+<% end -%>
+<% if @dnssec_enable -%>
+    dnssec-enable <%= @dnssec_enable %>;
+<% end -%>
+<% if @dnssec_validation -%>
+    dnssec-validation <%= @dnssec_validation %>;
+<% end -%>
+<% if @notify -%>
+    notify <%= @notify %>;
+<% end -%>
+
+<% if @include_localzones -%>
+<% unless [nil, :undefined, :undef, 'unmanaged'].include?(scope.lookupvar("::dns::localzonepath")) -%>
```

Comment:
Well, I am not satisfied with that either. $dns::localzonepath has a non-undef default value that one cannot unset via a parameter. However, I want to provide the possibility to manage RFC1912/RFC1918 zones via dns::zone (with and without `$enable_views`), in which case I don't want to include `$localzonepath`. Since `97b99cbf` until `fa16bad`, it was possible to achieve this by setting the `localzonepath` parameter to the empty string. Now, the empty string is not allowed any more, as it is not a valid Absolutepath. So I introduced a Variant Enum value called "unmanaged" instead. Maybe you can think of a more elegant way?